src/main/java/com/puppycrawl/tools/checkstyle/checks/design/FinalClassCheck.java

```diff
@@ -40,9 +40,34 @@
 
 /**
  * <p>
- * Checks that a class that has only private constructors and has no descendant
- * classes is declared as final.
+ * Check that identifies classes that can be declared as final are explicitly made final.
+ * The following is the type of classes that can be identified:
  * </p>
+ * <ol>
+ *   <li>
+ *       Private classes with implicit constructor.
+ *   </li>
+ *   <li>
+ *       Classes with all the type of modifier which contains only private constructor.
```

Comment:
I am not sure what you meant by "all the type of modifier". A class can't have ALL modifiers, but it can have "any" modifier. Is that what you meant?
Also is it really "only" a private constructor or is it actually contains "a" private constructor? I remember some test cases having public and private constructors, so the distinction is important.

If so to both, then: `Classes with all the type of modifier which contains only private constructor.` => `Classes with any modifier, which contains a private constructor.`

Reply:
>I think it is actually only, so maybe this is my mistake. It can contain multiple private constructors, right?

Yes, it can contain multiple private constructors.

Reply:
>Classes with any modifier, and contains only private constructors.

done